File: short_problems/Java/MaximizingXor.java
import java.util.Scanner;

public class MaximizingXor {
    public static int maximizeXor(int left, int right) {
        int xorred = left ^ right;
        int numBits = 0;
        while (xorred != 0) {
            xorred >>= 1;
            numBits ++;
        }
        return (1 << numBits) - 1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int left = sc.nextInt();
        int right = sc.nextInt();
        System.out.println(maximizeXor(left, right));
    }
}

File: short_problems/Python/maximizing_xor.py
#!/usr/local/bin/python3

def maximize_xor(left, right):
    xorred = left ^ right;
    num_bits = 0;
    while xorred != 0:
        xorred >>= 1
        num_bits += 1
    return (1 << num_bits) - 1

if __name__ == "__main__":
    left = int(input())
    right = int(input())
    print(maximize_xor(left, right))
